File: dagger2demo/src/main/java/com/cxsplay/dagger2demo/MainActivity.java
package com.cxsplay.dagger2demo;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.cxsplay.dagger2demo.bean.Pot;
import com.cxsplay.dagger2demo.daggermodule.DaggerMainActivityComponent;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    Pot pot;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DaggerMainActivityComponent.create().inject(this);


        findViewById(R.id.tv).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, pot.show(), Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: vmcoroutines/src/main/java/com/cxsplay/vmcoroutines/ui/MainActivity.kt
package com.cxsplay.vmcoroutines.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.cxsplay.vmcoroutines.R
import com.cxsplay.vmcoroutines.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var bind: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bind = DataBindingUtil.setContentView(this, R.layout.activity_main)
        init()
    }

    private fun init() {
        bind.btnList.setOnClickListener {
            startActivity(Intent(this, ProductsActivity::class.java))
        }
    }
}
